tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -236,6 +237,67 @@ def test_write_with_node_failures(self):
         ctx.assert_actions_triggered()
 
 
+class EndToEndCloudRetentionTest(EndToEndShadowIndexingBase):
+    segment_size = EndToEndShadowIndexingBase.segment_size // 4
+    retention_bytes = 10 * segment_size
+    topics = (TopicSpec(name=EndToEndShadowIndexingBase.s3_topic_name,
+                        partition_count=1,
+                        replication_factor=3,
+                        retention_bytes=retention_bytes,
+                        segment_bytes=segment_size), )
+
+    def __init__(self, test_context):
+        super().__init__(test_context,
+                         extra_rp_conf={
+                             'default_topic_replications': self.num_brokers,
+                             'cloud_storage_housekeeping_interval_ms': 1000 * 2
+                         })
+
+    @cluster(num_nodes=4, log_allow_list=CHAOS_LOG_ALLOW_LIST)
+    def test_retention_with_node_failures(self):
+        max_overshoot_percentage = 50
+
+        self.start_producer(throughput=10000)
+
+        def cloud_log_size() -> int:
+            s3_snapshot = S3Snapshot(self.topics, self.redpanda.s3_client,
+                                     self.s3_bucket_name, self.logger)
+            if not s3_snapshot.has_manifest_for_ntp(self.topic, 0):
+                self.logger.debug(f"No manifest present yet")
+                return 0
+
+            cloud_log_size = s3_snapshot.cloud_log_size_for_ntp(self.topic, 0)
+            ratio = cloud_log_size / self.retention_bytes
+            overshoot_percentage = max(ratio - 1, 0) * 100
+
+            self.logger.debug(f"Current cloud log size is: {cloud_log_size}")
+            self.logger.debug(f"Overshot by {overshoot_percentage}%")
+
+            if overshoot_percentage > max_overshoot_percentage:
+                raise RuntimeError(
+                    f"Cloud log size {overshoot_percentage}% greater than configured"
+                    f" retention (max allowed {max_overshoot_percentage}%)")
+
+            return cloud_log_size
+
+        with random_process_kills(self.redpanda) as ctx:
+            try:
+                wait_until(lambda: cloud_log_size() == -1,
+                           timeout_sec=90,
```

Comment:
How many process kills do we get in 90 seconds?  Maybe this should be longer and/or have the test assert that it did more than a certain number of kills, to avoid any subtle test bugs where if the kill/restore cycle is slow this test might just be doing one or two

Reply:
It only did three in the release run. I'll increase the time-out a bit and try to lower the interval between restarts a bit.

Reply:
I've increased the test length to 120 sec and made the action injector more aggressive.

Reply:
Test runs do about 5 process kills on average now. I'd say that's reasonable.